core/src/main/java/io/crate/core/collections/CollectionBucket.java

```diff
@@ -31,9 +32,19 @@
 public class CollectionBucket implements Bucket {
 
     private final Collection<Object[]> rows;
+    private final Function<Object[], Row> arrayToRowFunction;
 
     public CollectionBucket(Collection<Object[]> rows) {
+        this(rows, -1);
```

Comment:
instead of calling the other constructor and do an if check there, couldn't this one initialize the correct function directly?